vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/src/test/java/com/vaadin/flow/component/upload/tests/NonImmediateUploadIT.java

```diff
@@ -0,0 +1,50 @@
+package com.vaadin.flow.component.upload.tests;
+
+import java.io.File;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openqa.selenium.WebElement;
+
+import com.vaadin.flow.component.upload.testbench.UploadElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.testbench.TestBenchElement;
+
+@TestPath("vaadin-upload/non-immediate-upload")
+public class NonImmediateUploadIT extends AbstractUploadIT {
+
+    @Test
+    public void uploadMultipleFiles_shouldNotThrowException_onStart()
+            throws Exception {
+        uploadMultipleFiles_shouldNotThrowException("start-button");
+    }
+
+    @Test
+    public void uploadMultipleFiles_shouldNotThrowException_onRemove()
+            throws Exception {
+        uploadMultipleFiles_shouldNotThrowException("clear-button");
+    }
+
+    private void uploadMultipleFiles_shouldNotThrowException(String buttonType)
+            throws Exception {
+        open();
+        File file1 = createTempFile();
+        File file2 = createTempFile();
+        UploadElement upload = $(UploadElement.class).waitForFirst();
+        WebElement input = getInput(upload);
+        fillPathToUploadInput(input, file1.getPath(), file2.getPath());
+        WebElement button = getButton(upload, buttonType);
+        button.click();
+        TestBenchElement element = $("span").id("error-handler-message");
+        Assert.assertEquals("No errors", element.getText());
+    }
+
+    private WebElement getButton(UploadElement upload, String buttonType) {
```

Comment:
Had to see this visually to understand that it queries a button from a nested `vaadin-upload-file` element. Maybe the method name should reflect that it queries a button within an upload-file element, e.g. `getUploadFileElementButton`.

Reply:
Method was renamed